packages/wallet/hooks/useCreateAccounts.js

```diff
@@ -0,0 +1,19 @@
+import React, { useEffect } from 'react';
+import { extractAddressFromPassphrase } from '@wallet/utilities/account';
+import { generatePassphrase } from '@views/utilities/passphrase';
+
+const useCreateAccounts = () => {
+  const [accounts, setAccounts] = React.useState([]);
+
+  useEffect(() => {
+    const passphrases = [...Array(5)].map(generatePassphrase);
+    setAccounts(passphrases.map(pass => ({
+      address: extractAddressFromPassphrase(pass),
+      passphrase: pass,
+    })));
+  }, []);
+
+  return { setAccounts, accounts };
+};
+
```

Comment:
I think making accounts a dependency of the useEffect would trigger the useEffect  infinite times since `setAccounts` exists within the useEffect block. Also since `setAccounts` is just a setter function, adding it as a dependency kind of does nothing and even if `setAccounts` changes we would not want the useEffect to re-trigger.

Reply:
yes your are right, its just need setAccounts as dependencies because accounts it doesn't use here
but as see it just use for the getAccounts and it not updated or even can fetch new acounts it better to use usememo instend of setState
```
const useCreateAccounts = () =>   useMemo(() => {
   const passphrases = Array(5).fill("").map(() => {
       const  pass= generatePassphrase()
       return ({
      address: extractAddressFromPassphrase(pass),
      passphrase: pass,
    })
   });


```